src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,75 @@
+package pages;
+
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+
+import java.time.Duration;
+
+import static pages.LoginPage.*;
+
+
+public class BasePage {
+
+
+    public static WebDriver driver = null;
+
+
+    public static String url = null;
+
+    String playlistName = ":)";
+
+    public BasePage(WebDriver givenDriver) {
+        driver = givenDriver;
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        Actions actions = new Actions(driver);
+    }
```

Comment:
I recommend: make the variables protected, because protected access gives the subclass a chance to use the helper method or variable, while preventing a nonrelated class from trying to use it.

For example

```
    protected WebDriver driver;
    private WebDriverWait wait;
    protected Actions actions;

    public BasePage(WebDriver givenDriver) {
        driver = givenDriver;
        wait = new WebDriverWait(driver, Duration.ofSeconds(3));
        actions = new Actions(driver);
    }
```